test/async.ts

```diff
@@ -0,0 +1,66 @@
+import {
+    unprotect,
+    types,
+    addMiddleware
+    // TODO: export IRawActionCall
+} from "../src"
+import { test } from "ava"
+import { reaction } from "mobx"
+
+function delay<T>(time: number, value: T, shouldThrow = false): Promise<T> {
+    return new Promise((resolve, reject) => {
+        setTimeout(() => {
+            if (shouldThrow) reject(value)
+            else resolve(value)
+        }, time)
+    })
+}
+
+test.cb("can handle async actions", t => {
+    const Todo = types.model(
+        {
+            title: "get coffee"
+        },
+        {
+            *fetchData(kind: string) {
```

Comment:
Typings are an open issue, but beyond that is syntactically really nice. Especially because `fetchData: async(function* { ... })` is kinda ugly, but maybe that is the way forward for better typings. Really like the current syntax though :)